packages/node_modules/pouchdb-adapter-indexeddb/src/allDocs.js

```diff
@@ -101,57 +137,56 @@ export default function (idb, metadata, opts, callback) {
       }
     };
 
-    function include_doc(row, doc) {
-
-      row.doc = doc.data;
-      row.doc._id = doc.id;
-      row.doc._rev = doc.rev;
-
-      if (opts.conflicts) {
-        var conflicts = collectConflicts(doc);
-        if (conflicts.length) {
-          row.doc._conflicts = conflicts;
-        }
-      }
-
-      if (opts.attachments && doc.data._attachments) {
-        for (var name in doc.data._attachments) {
-          processing.push(processAttachment(name, doc, row.doc, opts.binary));
-        }
-      }
-    }
-
     var deleted = doc.deleted;
-
-    // TODO: I do not like this code
-    if (opts.deleted === 'ok') {
-      results.push(row);
-      if (deleted) {
+    if (deleted) {
+      if (keys) {
+        results.push(row);
         row.value.deleted = true;
         row.doc = null;
-      } else if (opts.include_docs) {
-        include_doc(row, doc);
       }
-    } else if (!deleted && skip-- <= 0) {
+    } else if (skip-- <= 0) {
       results.push(row);
       if (opts.include_docs) {
         include_doc(row, doc);
       }
       if (--limit === 0) {
-        return;
+        return false;
       }
     }
-    e.target.result.continue();
-  };
+    return true;
+  }
 
-  txn.oncomplete = function () {
+  function onTxnComplete() {
     Promise.all(processing).then(function () {
       callback(null, {
         total_rows: metadata.doc_count,
         offset: 0,
         rows: results
       });
     });
+  }
+
+  var cursor = descending ?
+    docStore.openCursor(keyRange, descending) :
+    docStore.openCursor(keyRange);
+
+  cursor.onsuccess = function (e) {
+
+    var doc = e.target.result && e.target.result.value;
+
+    // TODO: I have seen this before, but want to make sure I
+    // know exactly why its needed
```

Comment:
Honestly cant remember, it could be copied in from the oldest idb tutorials, if you can remove it and it doesnt fail tests it would be nice to get rid of

Reply:
Ok a quick test shows that removing that creates errors. Investigating...